vector/src/main/java/im/vector/app/core/utils/SystemSettingsProvider.kt

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2022 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.core.utils
+
+import android.content.Context
+import android.provider.Settings
+import javax.inject.Inject
+
+/**
+ * A helper to get system settings.
+ */
+interface SystemSettingsProvider {
+
+    /**
+     * @return system setting for font scale
+     */
+    fun getSystemFontScale(): Float
+}
+
+class SystemSettingsProviderImpl @Inject constructor(
```

Comment:
rather than `Impl` we could provide more context here by renaming the implementation `AndroidSystemSettingsProvider`, for other cases where the interface and implementation are the same (extracted for testing or design reasons) the convention is to use `Default{Interface}`

Reply:
I don't really think there is any other system rather than Android we could have, so `System` and `AndroidSystem` are quite same for me. And since so, I'll rename to `AndroidSystemSettingsProvider` :)